presto-common/src/test/java/com/facebook/presto/common/predicate/TestRange.java

```diff
@@ -281,4 +284,28 @@ public void testJsonSerialization()
         range = Range.lessThanOrEqual(DOUBLE, Double.MAX_VALUE);
         assertEquals(range, mapper.readValue(mapper.writeValueAsString(range), Range.class));
     }
+
+    public static final SqlFunctionProperties PROPERTIES = SqlFunctionProperties.builder().setTimeZoneKey(UTC_KEY).setLegacyTimestamp(true).setSessionStartTime(0).setSessionLocale(ENGLISH).setSessionUser("user").build();
+
+    @Test
+    public void testToString()
+    {
+        Range range = Range.all(VARCHAR);
+        assertEquals(range.toString(PROPERTIES), "(<min>, <max>)");
+
+        range = Range.equal(VARCHAR, utf8Slice("some string"));
+        assertEquals(range.toString(PROPERTIES), "[\"some string\"]");
+
+        range = Range.equal(VARCHAR, utf8Slice("here's a quote: \""));
+        assertEquals(range.toString(PROPERTIES), "[\"here's a quote: \\\"\"]");
+
+        range = Range.equal(VARCHAR, utf8Slice(""));
+        assertEquals(range.toString(PROPERTIES), "[\"\"]");
+
+        range = Range.greaterThanOrEqual(VARCHAR, utf8Slice("string with \"quotes\" inside")).intersect(Range.lessThanOrEqual(VARCHAR, utf8Slice("this string's quote is here -> \"")));
+        assertEquals(range.toString(PROPERTIES), "[\"string with \\\"quotes\\\" inside\", \"this string's quote is here -> \\\"\"]");
+
+        range = Range.greaterThan(VARCHAR, utf8Slice("string with \"quotes\" inside")).intersect(Range.lessThan(VARCHAR, utf8Slice("this string's quote is here -> \"")));
+        assertEquals(range.toString(PROPERTIES), "(\"string with \\\"quotes\\\" inside\", \"this string's quote is here -> \\\"\")");
+    }
```

Comment:
How about if if have field value as '<min>' or '<max>'? Please add a test for that.

Reply:
done